persistence/persistence-mybatis/src/test/java/org/cbioportal/persistence/mybatis/AlterationMyBatisRepositoryTest.java

```diff
@@ -0,0 +1,276 @@
+
+        List<AlterationCountByGene> result = alterationMyBatisRepository.getPatientAlterationCounts(
+            patientIdToProfileId, entrezGeneIds, mutationEventTypes, cnaEventTypes, QueryElement.PASS);
+
+        // For testSql.sql there are no more samples per patient for the investigated genes.
+        // Therefore, patient level counts are the same as the sample level counts.
+        Assert.assertEquals(3, result.size());
+        AlterationCountByGene result672 = result.stream().filter(r -> r.getEntrezGeneId() == 672).findFirst().get();
+        AlterationCountByGene result207 = result.stream().filter(r -> r.getEntrezGeneId() == 207).findFirst().get();
+        AlterationCountByGene result208 = result.stream().filter(r -> r.getEntrezGeneId() == 208).findFirst().get();
+        Assert.assertEquals((Integer) 5, result672.getTotalCount());
+        Assert.assertEquals((Integer) 4, result672.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 4, result207.getTotalCount());
+        Assert.assertEquals((Integer) 2, result207.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 2, result208.getTotalCount());
+        Assert.assertEquals((Integer) 2, result208.getNumberOfAlteredCases());
+    }
+
+    @Test
+    public void getSampleCnaCountLegacy() throws Exception {
+
+        // FIXME: whole CNA dedicated endpoint should be removed
+        entrezGeneIds = null; // only way it works; otherwise it tries to pair up geneids with alteration types with
+        List<CopyNumberCountByGene> result = alterationMyBatisRepository.getSampleCnaCounts(
+            sampleIdToProfileId, entrezGeneIds, cnaEventTypes);
+
+        Assert.assertEquals(3, result.size());
+        AlterationCountByGene result207up = result.stream().filter(r -> r.getEntrezGeneId() == 207 && r.getAlteration() == 2).findFirst().get();
+        AlterationCountByGene result207down = result.stream().filter(r -> r.getEntrezGeneId() == 207 && r.getAlteration() == -2).findFirst().get();
+        AlterationCountByGene result208up = result.stream().filter(r -> r.getEntrezGeneId() == 208).findFirst().get();
+        Assert.assertEquals((Integer) 1, result207up.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 1, result207down.getNumberOfAlteredCases());
+        Assert.assertEquals((Integer) 1, result208up.getNumberOfAlteredCases());
+    }
+    
+    @Test
+    public void getPatientCnaCountLegacy() throws Exception {
+
+        // FIXME: whole CNA dedicated endpoint should be removed
+        entrezGeneIds = null; // only way it works; otherwise it tries to pair up geneids with alteration types with
+        List<CopyNumberCountByGene> result = alterationMyBatisRepository.getPatientCnaCounts(
+            patientIdToProfileId, entrezGeneIds, cnaEventTypes);
+
+        // For testSql.sql there are no more samples per patient for the investigated genes.
+        // Therefore, patient level counts are the same as the sample level counts.
```

Comment:
It might be worthwhile to add a case with 2 samples belonging to the same sample in the mock test data ... it would give a chance here to put a test in place showing correct behavior for the logic / call chains distinguishing sample counts from patient counts.

Reply:
Ok, I made it:
```
       // FIXME: the CnaCountLegacy endpoint is different from the AlterationCount endpoint
        // because it returns a single additional value 'cytoband'. It would make sense to 
        // harmonize these endpoints (both or none return 'cytoband') and use the AlterationCount
        // endpoint for all counts. Discuss...
```